tests/conftest.py

```diff
@@ -557,11 +557,22 @@ def sm_source_dir(sm_entry_point, pytorch_training_script):
 
 @pytest.fixture(scope="session")
 def cluster(request):
-    """Parametrize over multiple fixtures - useful for running the same test on multiple hardware types."""
-    # Example: @pytest.mark.parametrize("cluster", [v100_gpu_cluster, k80_gpu_cluster], indirect=True)"""
     return request.getfixturevalue(request.param.__name__)
 
 
+@pytest.fixture(
+    params=[
+        "ondemand_cpu_cluster",
+        "v100_gpu_cluster",
+        "k80_gpu_cluster",
+        "a10g_gpu_cluster",
+    ],
+    ids=["cpu", "v100", "k80", "a10g"],
+)
+def ondemand_cluster(request):
+    return request.getfixturevalue(request.param)
```

Comment:
I don't think we need this, ideally I'd like to get rid of having to launch all these just to test the torch installs. We should try to break that test up to be more unity (or use mocks) so we can get rid of this, it's too crazy to launch all these gpus.